### PR TITLE
Revise example code

### PR DESCRIPTION
**Pull request by jklr**
Revised the example transit code 
- to be compliant with spec v1.0, 
- perform incremental TCP/UDP checksum updates, and
- against PSA architecture instead of v1model.

### CODE REVIEW

**Comment by jklr** [comment]
Pls ignore old commits, which have been already merged in previous PRs.
Will squash all commits when merge.

File: telemetry/specs/INT.mdk [hunk 1]
```diff
@@ -1018,8 +1018,12 @@ and it assumes INT over TCP as the encapsulation protocol.
 **This program is work in progress.**
```
Comment (mickeyspiegel) [inline telemetry/specs/INT.mdk:1031]: A few lines up it says "The program is written in P4_16 on [v1model architecture]"

File: telemetry/specs/INT.mdk [hunk 2]
```diff
@@ -1129,172 +1133,436 @@ header tcp_t {
...
+                hdr.ipv4.identification,
+                hdr.ipv4.flags, hdr.ipv4.fragOffset,
+                hdr.ipv4.ttl, hdr.ipv4.protocol,
+                hdr.ipv4.srcAddr,
+                hdr.ipv4.dstAddr
+            });
+        verify(hdr.ipv4.hdrChecksum == ck.get(), error.BadIPv4HeaderChecksum);
+
+        // For incremental update of TCP/UDP checksums
+        // subtract out the contributions of the IPv4 'psuedo header'
```
Comment (mickeyspiegel) [inline telemetry/specs/INT.mdk]: s/psuedo/pseudo

File: telemetry/specs/INT.mdk [hunk 3]
```diff
@@ -1129,172 +1133,436 @@ header tcp_t {
...
+                hdr.ipv4.flags, hdr.ipv4.fragOffset,
+                hdr.ipv4.ttl, hdr.ipv4.protocol,
+                hdr.ipv4.srcAddr,
+                hdr.ipv4.dstAddr
+            });
+            hdr.ipv4.hdrChecksum = ck.get();
+        }
+
+        // TCP/UDP header incremental checksum update.
+        // Restore the checksum state partilly calculated in the parser.
```
Comment (mickeyspiegel) [inline telemetry/specs/INT.mdk]: s/partilly/partially

File: telemetry/specs/INT.mdk [hunk 4]
```diff
@@ -1441,63 +1733,71 @@ control Int_egress(inout headers hdr, inout metadata meta,
...
-            } else {
+            if(hdr.int_header.remaining_hop_cnt == 0
+                    || hdr.int_header.e == 1) {
                 int_hop_cnt_exceeded();
+            } else {
+                int_prep.apply();
+                if (hdr.ipv4.totalLen + meta.int_metadata.insert_byte_cnt
+                        > meta.fwd_metadata.l3_mtu) {
+                    int_mtu_limit_hit();
+                } else {
```
Comment (mickeyspiegel) [inline telemetry/specs/INT.mdk]: Could add check for not yet defined instruction bits != 0 here

**Review by mickeyspiegel — APPROVED** [review]
I don't know much about psa, so just high level comments here

File: telemetry/specs/INT.mdk (hunk 1 above)
Comment (jklr) [reply telemetry/specs/INT.mdk:1031]: Done.

**Pull request merged**